osquery/tables/system/darwin/unified_log.mm

```diff
@@ -0,0 +1,225 @@
+
+const std::map<std::string, std::string> kColumnToOSLogEntryProp = {
+    {"timestamp", "date"},
+    {"message", "composedMessage"},
+    {"storage", "storeCategory"},
+    {"activity", "activityIdentifier"},
+    {"process", "process"},
+    {"pid", "processIdentifier"},
+    {"sender", "sender"},
+    {"tid", "threadIdentifier"},
+    {"subsystem", "subsystem"},
+    {"category", "category"}};
+
+const std::map<std::string, bool> kColumnIsNumeric = {{"timestamp", false},
+                                                      {"message", false},
+                                                      {"storage", true},
+                                                      {"activity", true},
+                                                      {"process", false},
+                                                      {"pid", true},
+                                                      {"sender", false},
+                                                      {"tid", true},
+                                                      {"subsystem", false},
+                                                      {"category", false}};
+
+std::string convertLikeExpr(const std::string& value) {
+  // for the LIKE operator in NSPredicates, '*' matches 0 or more characters
+  //   and '?' matches a single character
+  std::string res = ba::replace_all_copy(value, "%", "*");
+  ba::replace_all(res, "_", "?");
+  return res;
+}
+
+void addQueryOp(NSMutableArray* preds,
+                const std::string& key,
+                const std::string& value,
+                ConstraintOperator op) {
+  if (kSupportedOps.count(op) > 0 && kColumnToOSLogEntryProp.count(key) > 0) {
+    std::string modified_val = value;
+    std::string modified_key = kColumnToOSLogEntryProp.at(key);
+    if (op == LIKE) {
+      modified_val = convertLikeExpr(value);
+    }
+    NSExpression* keyExp = [NSExpression
+        expressionForKeyPath:[NSString
+                                 stringWithUTF8String:modified_key.c_str()]];
+    NSString* valStr = [NSString stringWithUTF8String:modified_val.c_str()];
+
+    NSExpression* valExp = nil;
+    if (key == "timestamp") {
+      double provided_timestamp = [valStr doubleValue];
+      valExp = [NSExpression
+          expressionForConstantValue:
+              [NSDate dateWithTimeIntervalSince1970:provided_timestamp]];
+    } else if (kColumnIsNumeric.at(key)) {
+      valExp =
+          [NSExpression expressionWithFormat:@"%lld", [valStr longLongValue]];
+    } else {
+      valExp = [NSExpression expressionForConstantValue:valStr];
+    }
+
+    NSPredicate* pred = [NSComparisonPredicate
+        predicateWithLeftExpression:keyExp
+                    rightExpression:valExp
+                           modifier:NSDirectPredicateModifier
+                               type:kSupportedOps.at(op)
+                            options:0];
+    [preds addObject:pred];
+  }
+}
+
+QueryData genUnifiedLog(QueryContext& queryContext) {
```

Comment:
A mandatory `WHERE` clause should be implemented to limit the scope of the query and avoid watchdog issues with simple `SELECT * FROM table` statements